src/extractor/guidance/sliproad_handler.cpp

```diff
@@ -344,7 +344,20 @@ operator()(const NodeID /*nid*/, const EdgeID source_edge_id, Intersection inter
         }
 
         // If the sliproad candidate is a through street, we cannot handle it as a sliproad.
-        if (isThroughStreet(sliproad_edge, target_intersection))
+        // std::cout << "sliproad_edge: " << sliproad_edge << std::endl;
+        auto index_of_sliproad = [&](const auto &road) {
+            // std::cout << "road.eid: " << road.eid << std::endl;
+            return road.eid == sliproad_edge;
+        };
+        auto index = std::find_if(begin(intersection), end(intersection), index_of_sliproad) -
+                     begin(intersection);
```

Comment:
What if `index_of_sliproad` is never true? I think index should be out of bounds then.

Reply:
So I introduced https://github.com/Project-OSRM/osrm-backend/pull/4751#discussion-diff-161700751R36 to deal with this case. Is ^ not how I should deal with this case?